src/storage-types/src/sources.rs

```diff
@@ -1199,8 +1201,8 @@ impl PartDecoder<SourceData> for SourceDataDecoder {
                     .expect("error should be valid");
                 val.0 = Err(err);
             }
-            (true, Some(_)) | (false, None) => {
-                panic!("SourceData should have exactly one of ok or err")
+            x @ (true, Some(_)) | x @ (false, None) => {
+                panic!("SourceData should have exactly one of ok or err, {idx} @ {x:?}\n{self:?}")
```

Comment:
We may want to crank this back slightly... IIRC we don't want user-level errors in our logs.

Reply:
Makes sense! I added this originally when doing some of the debugging related to `arrow2`, removed